src/directions/main.ts

```diff
@@ -672,42 +695,51 @@ export default class MaplibreGlDirections {
 
     this.assignWaypointsCategories();
 
+    const waypointEvent = new MaplibreGlDirectionsWaypointEvent("addwaypoint", undefined);
+    this.fire(waypointEvent);
+
     this.draw();
-    await this.fetchDirections();
+    await this.fetchDirections(waypointEvent);
   }
 
   /**
    * Adds a waypoint at the specified coordinates to the map and re-fetches the routes.
    *
-   * @param {[number, number]} waypoint The coordinates at which the waypoint should be added
-   * @param {number?} index The index the waypoint should be inserted at. If omitted, the waypoint is inserted at the end
-   * @return {Promise<void>} Resolved after the routing request has finished
+   * @param waypoint The coordinates at which the waypoint should be added
+   * @param index The index the waypoint should be inserted at. If omitted, the waypoint is inserted at the end
+   * @return Resolved after the routing request has finished
    */
-  async addWaypoint(waypoint: [number, number], index?: number) {
+  async addWaypoint(waypoint: [number, number], index?: number, originalEvent?: MapMouseEvent | MapTouchEvent) {
```

Comment:
Document originalEvent in jsdoc

Reply:
@wipfli it left un-documented specifically. The end user shouldn't be using this parameter, it's only for internal-purposes. I thought of adding another wrapper-method with that third parameter, but I think it'd be an overkill.

Reply:
Interesting thanks. In that case, please add a comment.

Reply:
Something like "for internal use only"